src/box/memtx_engine.cc

```diff
@@ -1457,7 +1498,11 @@ memtx_engine_join(struct engine *engine, struct engine_join_ctx *arg,
 static void
 memtx_engine_complete_join(struct engine *engine, struct engine_join_ctx *arg)
 {
-	(void)engine;
+	if (arg->cursor != NULL) {
+		memtx_engine_complete_checkpoint_join(engine, arg);
+		return;
```

Comment:
Nit: In C it's possible to write this in one line, like in `memtx_engine_prepare_join`:

```C
	if (arg->cursor != NULL)
		return memtx_engine_complete_checkpoint_join(engine, arg);
```

In `vinyl_engine_complete_join`, too.